README.md

```diff
@@ -30,40 +36,65 @@ function LoginForm() {
 }
 ```
 
-### Styling
+### How to style components?
 
 Every component holds a minimum amount of CSS. You can modify them via the CSS variables. See `_colors.scss` and `_measurement.scss`
 
-Here is a simple example that shows how to customize `Button` and `Input`
+Here is a simple example that shows how to customize `Button` and `Input` styles by overriding the default CSS variables:
+
+```scss
+.button {
+  // Override the default button styles using CSS variables
 
-```css
-// _button.css
-.primary-button {
   --button-bg: #989898;
   --button-color: black;
 }
 
-// _input.css
 .input {
+  // Override the default input styles using CSS variables
+
   --default-border-color: black;
 }
 ```
 
+## Development
+
+[Storybook](#storybook) is suggested for the development environment. It allows you to see the components in isolation and interact with them. It also supports hot-reloading, i.e. when you change the component, it automatically reloads the component in the browser.
+
+First of all, you need to install the dependencies, in the project root folder, run:
+
+```bash
+npm install
+```
+
+> ⚠️ Make sure you are using the exact version of `node` and `npm` that are specified in the `engines` field of [package.json](/package.json) file. Otherwise, you may face some unexpected issues.
+
 ### Storybook
 
-- To run Storybook `npm run storybook`
-- To generate Storybook build `npm run storybook:build`
+Storybook is a development environment for UI components. It allows you to browse a component library, view the different states of each component, and interactively develop and test components.
+
+To run the Storybook development server on your local environment, you can use the following command:
+
+```bash
+npm run storybook
+```
+
+To generate a static build of the Storybook (usually, you don't need this. This is only necessary when you want to publish it to somewhere), you can use the following command:
 
-### Development
+```bash
+npm run storybook:build
+```
 
-For recommended `node` and `npm` versions, you can check `package.json`
+### Production Build
 
-You can start to development with `npm run dev` command. The command watches for changes and builds the toolkit. If you want to generate a production ready build you can use `npm run build`.
+The production deployment is automated by GitHub Actions. Check the [.github/workflows/new-version.yml](/.github/workflows/new-version.yml) file for more information.
 
-Or you can run `npm run storybook` to see the components live. Storybook has own Webpack config that compiles and runs the components.
+---
 
-### Linter
+If you need to generate a production ready build for some reason, use:
```

Comment:
Maybe we can also briefly tell our `pre-commit` process and husky.

Reply:
👍 Yes we can mention it, I'll do it after moving to bakers